### PR TITLE
 App Cache instance Object change

### PR DESCRIPTION
**Pull request by ankit-privado**

### CODE REVIEW

File: src/main/scala/ai/privado/entrypoint/Main.scala [hunk 1]
```diff
@@ -34,11 +35,11 @@ object Main {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
   def main(args: Array[String]): Unit = {
-
+    val appCache = new AppCache()
```
Comment (ankit-privado) [inline src/main/scala/ai/privado/entrypoint/Main.scala:38]: @khemrajrathore pls confirm this change in `main.scala`

File: src/main/scala/ai/privado/entrypoint/ScanProcessor.scala [hunk 2]
```diff
@@ -332,19 +332,19 @@ object ScanProcessor extends CommandProcessor {
...
     ruleCache.setRule(processedRules)
     ruleCache
   }
 
-  private def processCpg() = {
+  private def processCpg(appCache: AppCache) = {
     val sourceRepoLocation = File(config.sourceLocation.head).path.toAbsolutePath.toString.stripSuffix("/")
     // Setting up the application cache
-    AppCache.init(sourceRepoLocation)
+    // val appCache = new AppCache(sourceRepoLocation)
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:347]: where are we setting `sourceRepoLocation`?

Comment (ankit-privado) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala]: yes, there only https://github.com/Privado-Inc/privado-core/pull/1023/files#diff-47c8acbc2ffafd950e96bdaf4e0b8d26671f59626a039824e12b051932d422ebR347.

**Comment by khemrajrathore** [comment]
@ankit-privado Looking good, please do a `dev` sync with this branch

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**